ports/wakingmars/port.json

```
@@ -12,7 +12,7 @@
             "Cebion"
         ],
         "desc": "Waking Mars is a one-of-a-kind adventure in which you jetpack through subterranean Mars encountering a host of extraordinary alien lifeforms that operate as a living, breathing ecosystem.",
-        "inst": "Add your Linux (humble) game files to the gamedata folder\n\nFile structure:\n/roms/ports/wakingmars/gamedata/wakingmars_files\n/roms/ports/wakingmars/gamedata/wakingmars.png",
+        "inst": "Add your Linux (humble) game files from the WakingMars-1.2.1-Linux.tar.gz tarball to the /roms/wakingmars/gamedata folder (you will need to create the gamedata folder if it does not exist).\n\nFile structure:\n/roms/ports/wakingmars/gamedata/GameData/\n/roms/ports/wakingmars/gamedata/lib/\n/roms/ports/wakingmars/gamedata/wakingmars\n/roms/ports/wakingmars/gamedata/wakingmars.png\n\nOn some platforms you may need to run the port script (/roms/ports/Waking Mars.sh) from Options->File Manager in order to prevent the game from segfaulting whilst it starts.",
```

```suggestion
        "inst": "Add your Linux (humble) game files from the WakingMars-1.2.1-Linux.tar.gz tarball to the /roms/wakingmars/gamedata folder .",
```
We'll add the gamedata folder to the port.
I'm afraid we can't add pictures to the instructions field. 
In newer port.json variants we can add markdown instructions though 

See https://portmaster.games/port-json.html

As  for the "n some platforms you may need to run the port script (/roms/ports/Waking Mars.sh) from Options->File Manager in order to prevent the game from segfaulting whilst it starts." What is the reason behind this? I don't know of a single port that has this problem, we'd have to find out the reason why this behaves like this and on what cfw this is the case.

---

> As for the "On some platforms you may need to run the port script (/roms/ports/Waking Mars.sh) from Options->File Manager in order to prevent the game from segfaulting whilst it starts." What is the reason behind this? I don't know of a single port that has this problem, we'd have to find out the reason why this behaves like this and on what cfw this is the case.

This is on an R36S running CFW_NAME=ArkOS AeUX ( https://github.com/AeolusUX/ArkOS-R3XS ), with very many games installed (resulting in EmulationStation using about 65% of system RAM, which may be relevant).

Attempting to launch from Ports->Waking Mars results in a segfault:
cat: /home/ark/.config/.OS: No such file or directory
BOX86 using "/roms2/ports/wakingmars/gl4es/libGL.so.1" as libGL.so.1
Dynarec for ARMv8, with extension: HALF FAST_MULT EDSP NEON VFPv4 IDIVA AES PMULL PageSize:4096 [GPTK]: Running in UINPUT output mode.
Running on Cortex-A35 with 4 Cores
BOX86_LD_LIBRARY_PATH: /roms2/ports/wakingmars/box86/x86/:/roms2/ports/wakingmars/box86/native/:/roms2/ports/wakingmars/libs/x86/
Using default BOX86_PATH: ./:bin/
Counted 33 Env var
BOX86 try to Preload /usr/$LIB/libSDL2-2.0.so.0.18.2 
Looking for gamedata/wakingmars
Rename process to "wakingmars"
Warning, cannot pre-load /usr/$LIB/libSDL2-2.0.so.0.18.2
Using native(wrapped) libSDL-1.2.so.0
Using native(wrapped) libm.so.6
Using native(wrapped) libdl.so.2
Using native(wrapped) libc.so.6
Using native(wrapped) ld-linux.so.2
Using native(wrapped) libpthread.so.0
Using native(wrapped) librt.so.1
Using native(wrapped) libbsd.so.0
Using emulated /roms2/ports/wakingmars/libs/x86/libfreeimage.so.3
Using native(wrapped) libalut.so.0
Using native(wrapped) libopenal.so.1
Using emulated /roms2/ports/wakingmars/gamedata/lib/libsteam_api.so
Using emulated /roms2/ports/wakingmars/gamedata/lib/libchipmunk.so.6.1.1
LIBGL: Initialising gl4es
LIBGL: v1.1.5 built on Jun 30 2022 18:47:25
LIBGL: using GBM
LIBGL: Using GLES 2.0 backend
LIBGL: loaded: libGLESv2.so
LIBGL: loaded: libEGL.so
LIBGL: loaded: libgbm.so
LIBGL: loaded: libdrm.so.2
LIBGL: Using GLES 2.0 backend
LIBGL: GBM on card /dev/dri/card0 is Available
LIBGL: Hardware Full NPOT detected and used
LIBGL: Extension GL_EXT_blend_minmax  detected and used
LIBGL: FBO are in core, and so used
LIBGL: PointSprite are in core, and so used
LIBGL: CubeMap are in core, and so used
LIBGL: BlendColor is in core, and so used
LIBGL: Blend Subtract is in core, and so used
LIBGL: Blend Function and Equation Separation is in core, and so used
LIBGL: Texture Mirrored Repeat is in core, and so used
LIBGL: Extension GL_OES_mapbuffer  detected
LIBGL: Extension GL_OES_element_index_uint  detected and used
LIBGL: Extension GL_OES_packed_depth_stencil  detected and used
LIBGL: Extension GL_OES_depth24  detected and used
LIBGL: Extension GL_OES_rgb8_rgba8  detected and used
LIBGL: Extension GL_EXT_texture_format_BGRA8888  detected and used
LIBGL: Extension GL_OES_depth_texture  detected and used
LIBGL: Extension GL_OES_texture_stencil8  detected and used
LIBGL: Extension GL_EXT_texture_rg  detected and used
LIBGL: Extension GL_EXT_color_buffer_float  detected and used
LIBGL: Extension GL_EXT_color_buffer_half_float  detected and used
LIBGL: high precision float in fragment shader available and used
LIBGL: Max vertex attrib: 16
LIBGL: Extension GL_OES_standard_derivatives  detected and used
LIBGL: Max texture size: 8192
LIBGL: Max Varying Vector: 15
LIBGL: Texture Units: 16/16 (hardware: 16), Max lights: 8, Max planes: 6
LIBGL: Max Color Attachments: 1 / Draw buffers: 1
LIBGL: Hardware vendor is ARM
LIBGL: GLSL 300 es supported
LIBGL: GLSL 310 es supported and used
LIBGL: sRGB surface supported
LIBGL: EGLImage to Texture2D supported
LIBGL: EGLImage to RenderBuffer supported
LIBGL: Targeting OpenGL 2.1
LIBGL: Not trying to batch small subsequent glDrawXXXX
LIBGL: try to use VBO
LIBGL: glXMakeCurrent FBO workaround enabled
LIBGL: FBO workaround for using binded texture enabled
LIBGL: Force texture for Attachment color0 on FBO
LIBGL: Hack to trigger a SwapBuffers when a Full Framebuffer Blit on default FBO is done
LIBGL: glX Will try to recycle EGL Surface
LIBGL: Current folder is:/roms2/ports/wakingmars
Using native(wrapped) libGL.so.1
Using emulated /roms2/ports/wakingmars/gamedata/lib/libboost_filesystem.so.1.46.1
Using emulated /roms2/ports/wakingmars/gamedata/lib/libboost_system.so.1.46.1
Using native(wrapped) libfreetype.so.6
Using native(wrapped) libvorbisfile.so.3
Using emulated /roms2/ports/wakingmars/box86/x86/libstdc++.so.6
Using emulated /roms2/ports/wakingmars/box86/x86/libgcc_s.so.1
[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
[S_API FAIL] SteamAPI_Init() failed; unable to locate a running instance of Steam, or a local steamclient.dll.
Failed to create secure directory (/run/user/1002//pulse): No such file or directory
******** WAKING MARS - VERSION 1.2.1 **********************************************************************
Environment variables SDL_VIDEO_FULLSCREEN_DISPLAY, SDL_VIDEO_FULLSCREEN_HEAD not set. Setting to '0' to force single monitor.2619|SIGSEGV @0xebeb7d04 (???(0xebeb7d04)) (x86pc=0x81c13bb//roms2/ports/wakingmars/gamedata/wakingmars:"/roms2/ports/wakingmars/gamedata/wakingmars/_ZN7GameApp16InitializeOpenGLEv + 15", esp=0xf67b57b0, stack=0xf5fb6000:0xf67b6000 own=(nil) fp=0xf67b5888), for accessing 0xc (code=1/prot=0), db=0xebfd04b8(0xebeb7cfc:0xebeb7dc8/0x81c13b5:0x81c13fb//roms2/ports/wakingmars/gamedata/wakingmars/_ZN7GameApp16InitializeOpenGLEv + 15:clean, hash:5bb7166f/5bb7166f)
EAX:0x00000000 ECX:0x00000000 EDX:0x65560a50 EBX:0xf67b591c 
ESP:0xf67b57b0 EBP:0xf67b5888 ESI:0x00000001 EDI:0x00000000 
ESP-0x10:0x6556077c ESP-0x0c:0x63f4d018 ESP-0x08:0x00000740 ESP-0x04:0x081c13b5
ESP+0x00:0x00000000 ESP+0x04:0x65560a5c ESP+0x08:0x655609dc ESP+0x0c:0x655609c4
2619|Double SIGSEGV (code=1, pc=0xebeb7d04, addr=0xc)!
Stop waiting for remaining thread 2635
Stop waiting for remaining thread 2636
Stop waiting for remaining thread 2637
Stop waiting for remaining thread 2638
Stop waiting for remaining thread 2639
Stop waiting for remaining thread 2640
Stop waiting for remaining thread 2641
... (4 lines left)

---

We'll investigate.

---

Added a PR for the missing folder.
https://github.com/PortsMaster/PortMaster-New/pull/1347

---

> I'm afraid we can't add pictures to the instructions field.

The mention of /roms/ports/wakingmars/gamedata/wakingmars.png is _not_ a picture for the instructions field, but was _already in_ the instructions, and refers to a file that they say to copy from the Linux tarball to the gamedata folder.

---

> > I'm afraid we can't add pictures to the instructions field.
> 
> The mention of /roms/ports/wakingmars/gamedata/wakingmars.png is _not_ a picture for the instructions field, but was _already in_ the instructions, and refers to a file that they say to copy from the Linux tarball to the gamedata folder.

I see that was an error in the first place so can be removed.